src/carousel.tsx

```diff
@@ -457,6 +464,24 @@ const BaseCarousel: React.FC<BaseCarouselProps> = ({
 
     const shouldAutoplay = useShouldAutoplay(!!autoplay, carouselRef);
 
+    React.useEffect(() => {
+        if (
+            initialActiveItem &&
+            (initialActiveItem !== initialItemInfo.current.initialItem ||
+                itemsPerPageFloor !== initialItemInfo.current.itemsPerPage)
+        ) {
+            goToPage(Math.floor(initialActiveItem / itemsPerPageFloor));
+            initialItemInfo.current.initialItem = initialActiveItem;
+            initialItemInfo.current.itemsPerPage = itemsPerPageFloor;
+        }
+    }, [
+        initialActiveItem,
+        itemsPerPageFloor,
+        goToPage,
+        initialItemInfo.current.initialItem,
+        initialItemInfo.current.initialItem,
+    ]);
+
```

Comment:
I'm probably missing something but if you want to react to the `initialActiveItem` change, the ref is not needed, you just need an effect like:
```js
React.useEffect(() => {
    goToPage(Math.floor(initialActiveItem / itemsPerPageFloor));
}, [initialActiveItem, goToPage, itemsPerPageFloor]);
```

But I think you shouldn't react to `initialActiveItem` changes. This value should be used only for the initial render and ignore changes. Then you should use a ref and use its value to set the initial position. Don't react to changes.

To update the storybook carousel when changing the `arg`, perhaps you want to set a `key` value to the carousel to remount it when the `initialActiveItem` changes

Reply:
I need to react to itemsPerPageFloor because the carousel renders with 1 in this variable the first time and then it re'renders with the correct value. so I need to add dependencies array, so I need to add all dependencies to make the warning disappear.
Also if the prop changes, its desirable to move to the page the prop reffers to (in account there is a use case in the header of the invoicing page (if the selected invoice changes, the carousel must go to the page in where the invoice is present).

Reply:
Hook changed to what you suggested in the other comment ;)